src/libdrakvuf/libdrakvuf.h

```diff
@@ -274,6 +302,32 @@ char *drakvuf_get_current_process_name(drakvuf_t drakvuf,
                                        uint64_t vcpu_id,
                                        x86_registers_t *regs);
 
+
+bool drakvuf_get_current_thread_id( drakvuf_t drakvuf, 
+                                    uint64_t vcpu_id, 
+                                    x86_registers_t *regs,
+                                    uint32_t *thread_id );
+
+bool drakvuf_get_previous_mode( drakvuf_t drakvuf, 
```

Comment:
What would you think of 2 functions one for current thread (mostly used) and the other one for generic thread?

Kindof...
```
bool drakvuf_get_thread_previous_mode( drakvuf_t drakvuf, addr_t kthread, privilege_mode_t *previous_mode )
{
    if ( kthread )
    {
        if ( vmi_read_8_va( drakvuf->vmi, kthread + offsets[ KTHREAD_PREVIOUSMODE ], 0,
                            (uint8_t *)previous_mode ) == VMI_SUCCESS )
            return true ;
    }

    return false ;
}

bool drakvuf_get_current_thread_previous_mode( drakvuf_t drakvuf, drakvuf_trap_info_t *info,
                                               privilege_mode_t *previous_mode )
{
    addr_t kthread = drakvuf_get_current_thread( drakvuf, info->vcpu, info->regs );

    return drakvuf_get_thread_previous_mode( drakvuf, kthread, previous_mode );
}
```